src/gradata/enhancements/rule_canary.py

```
@@ -0,0 +1,301 @@
+            from events import emit
+            emit(
+                "RULE_ROLLBACK",
+                "rule_canary:rollback_rule",
+                {
+                    "rule_category": rule_category,
+                    "reason": reason,
+                    "rollback_confidence": ROLLBACK_CONFIDENCE,
+                },
+                tags=[f"category:{rule_category}", "canary:rollback"],
+            )
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)